### PR TITLE
Feature/tao 6225 adapt lti user service

### PR DESCRIPTION
**Pull request by siwane**

### CODE REVIEW

File: models/classes/user/KvLtiUserService.php [hunk 1]
```diff
@@ -66,11 +66,22 @@ protected function getPersistence()
...
+        $technicalId = self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri();
+
+        if (empty($user->getIdentifier())) {
+            $user->setIdentifier($technicalId);
+        }
+
+        $taoUserId = $user->getIdentifier();
+
         $this->getPersistence()->set(
             self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri(),
```
Comment (ionutpad) [inline models/classes/user/KvLtiUserService.php]: You can replace the entire `self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri()` with ` $technicalId` I guess

File: models/classes/user/KvLtiUserService.php [hunk 2]
```diff
@@ -92,7 +103,17 @@ public function getUserIdentifier($ltiUserId, $consumer)
      */
     public function getUserDataFromId($taoUserId)
     {
-        $data = $this->getPersistence()->get($taoUserId);
+        if (!$this->getPersistence()->exists(self::LTI_USER_LOOKUP . $taoUserId)) {
+            if ($this->getPersistence()->exists($taoUserId)) {
+                $this->getPersistence()->set(self::LTI_USER_LOOKUP . $taoUserId, $taoUserId);
+                $data = $this->getPersistence()->get($taoUserId);
+            } else {
+                return false;
```
Comment (ionutpad) [inline models/classes/user/KvLtiUserService.php]: You are breaking the method definition by returning null, in the abstract it's saying that it's an array, and in the `OntologyLtiUserService` it's returning  null, can you return null as well.

Comment (siwane) [reply models/classes/user/KvLtiUserService.php]: hum, `You are breaking the method definition by returning null` versus `can you return null as well`

File: models/classes/user/KvLtiUserService.php (hunk 1 above)
Comment (siwane) [reply models/classes/user/KvLtiUserService.php]: done

**Pull request merged**